### DISCUSSION TITLE
Steam (included in image)

### DISCUSSION THREAD

**Discussion by martinbone** [Joshua-Riek/ubuntu-rockchip#73] [Ideas]
This is a great project. Are you planning on including Steam at some point?

**Comment by Joshua-Riek** [comment]
I was thinking about this, I would love to include steam in the image. Setting up box86, box64, wine, and steam from my testing can be complicated, and users can run into issues. I still have not set it up correctly.

But before adding steam or anything, I need to make sure it's stable, I don't want to release something buggy. This is one of the reasons why I have not included Kodi by default. So at the very least, I'll put up a guide on the wiki or create a script to install steam properly.

**Comment by Fredobedo** [comment]
And here is an interesting list of games running natively on ARM: https://www.pcgamingwiki.com/wiki/List_of_Windows_ARM_games
Maybe some of them can be installed with Steam or be installed another way.

**Comment by martinbone** [comment]
Ok. I guess it's still really early days. Even a guide on how to install
Steam (at some point in time) would be really excellent. Thanks.

**Comment by HSTe** [comment]
I guess fex-emu could also be a solution to run steam on this device

**Reply by Fredobedo** [reply]
From what I see here https://youtu.be/B3SOMWdFxBw 
It is possibly a solution but at a performance cost, box64 being the most efficient. (Apart from native)

**Comment by HSTe** [comment]
I tried install steam after compiling and installing box64 and box86. Running the ./install_steam.sh in box86 failed because of the panfork mesa packages.
The following packages have unmet dependencies:
 libgl1-mesa-dri : Breaks: libgl1-mesa-dri:armhf (!= 23.0~panfork~csf~git221210.120202c6757~j+3) but 22.2.5-0ubuntu0.1~22.04.1 is to be installed
 libgl1-mesa-dri:armhf : Recommends: libgl1-amber-dri:armhf but it is not going to be installed
                         Breaks: libgl1-mesa-dri (!= 22.2.5-0ubuntu0.1~22.04.1) but 23.0~panfork~csf~git221210.120202c6757~j+3 is to be installed
 libglapi-mesa : Breaks: libglapi-mesa:armhf (!= 23.0~panfork~csf~git221210.120202c6757~j+3) but 22.2.5-0ubuntu0.1~22.04.1 is to be installed
 libglapi-mesa:armhf : Breaks: libglapi-mesa (!= 22.2.5-0ubuntu0.1~22.04.1) but 23.0~panfork~csf~git221210.120202c6757~j+3 is to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.

**Reply by Joshua-Riek** [reply]
I have several held packages with apt for stability and version control reasons. I plan on creating a ppa mirror to fix this so this problem will not occur. 

However, if you have amazingfate's ppa already you can try `sudo apt-mark unhold $(apt-mark showhold)`. As a word of warning, this could break hardware acceleration, multiple packages I have held for a reason, and removing the hold could produce issues.

**Reply by HSTe** [reply]
I tried it and then the steam installed, but the desktop was removed.
I installed gnome again and then steam started. Tried glmark2 and it used llvm
Installed the panfork ppa and upgraded  and now glmark used panfork .
Tried starting steam and it crashed on the login-page. I guess we need a panfork repo with the same armhf buildt packages

**Reply by Joshua-Riek** [reply]
The next release will include a mirror of the panfork ppa. This will allow me to test and control package changes, so I don't have to worry about an updated package from panfork breaking things. Steam should be much easier to get working, then. I still want to create a script for installing it but I have been busy with OS optimizations.
